### DISCUSSION TITLE
What is the difference between using diffusers and Fooocus to generate realistic images?

### DISCUSSION THREAD

**Discussion by kukovinetsyevhenii** [lllyasviel/Fooocus#3341] [Q&A] [answered]
When I first try to generate images using  stable diffusion, perhaps, looks easy, went to chatgpt and saw piece of code they are giving to me. But they are not working.
What I found is they are using diffusers.
I want to know the main differnce between diffusers or fooocus
Then Can we use diffusers to generate images then how?
Appreciate your help.

**Comment by mashb1t** [comment, marked as answer]
Fooocus and diffusers are two completely separate open source projects which are only related in their purpose to create images.

Diffusers is a python package which offers functionality to load models and create pipelines to use in other software, not necessarily by itself (minimum is that you have to create a python file and call diffusers). It's an abstraction layer for more complex code and best used for specific individual needs.

Fooocus is an opinionated software able to run independently and using multiple packages and workflows to generate best results with as little as possible effort. It does not use diffusers but ldm as backbone, incl. influences of ComfyUI, A1111 and many more.

Some Fooocus forks do use diffusers in parts of their workflows, but Fooocus does not (and 99% never will), which is also why Fooocus is more optimized in terms of hardware needs.

Hope this helps in clarifying the differences.

**Reply by oldhand7** [reply]
Could you tell me how to test wit diffusers?
I want to compare them myself
Thanks.

**Reply by mashb1t** [reply]
- https://huggingface.co/docs/diffusers/index
- https://huggingface.co/docs/diffusers/tutorials/tutorial_overview
- https://huggingface.co/docs/diffusers/quicktour

**Reply by mashb1t** [reply]
also see "Usability over Performance" in https://huggingface.co/docs/diffusers/conceptual/philosophy#usability-over-performance
